jaeger-core/src/test/java/io/jaegertracing/internal/propagation/TraceContextCodecTest.java

```diff
@@ -160,10 +164,31 @@ public void testEmptyTraceStateNotPropagated() {
     assertEquals(EXAMPLE_TRACE_PARENT, injectCarrier.get(TRACE_PARENT));
   }
 
+  @Test
+  public void testDebugIdWithoutTraceHeader() {
+    Map<String, String> extractCarrier = new HashMap<>();
+    TextMapAdapter textMap = new TextMapAdapter(extractCarrier);
+    textMap.put(Constants.DEBUG_ID_HEADER_KEY, EXAMPLE_DEBUG_ID);
+    JaegerSpanContext spanContext = traceContextCodec.extract(textMap);
+    assertTrue(spanContext.isDebug());
```

Comment:
the context itself is not "debug". Instead, you can check that it contains the correlation ID.